### DISCUSSION TITLE
How to reproduce transformation on other data?

### DISCUSSION THREAD

**Discussion by studywolf** [theochem/procrustes#200] [Q&A] [answered]
Hello, 

I would like to find the transformation from A to B, and then apply those some transformation steps to a matrix C. I've been looking through the code, and I was expecting to do something like 
`np.dot((C + centroid_a) * scaling_factor_a, results.t) / scaling_factor_b - centroid_b` 
where the `centroid` and `scaling_factor` are calculated using the `_translate_array` and `_scale_array` functions with input `(A, None)`. But I'm not successfully even reproducing `results.new_a` when I do the `(A + centroid) * scaling_factor`. Can anyone lend some insight into where I'm misunderstanding things?

**Comment by studywolf** [comment, marked as answer]
Nevermind, resolved! 
```
import matplotlib.pyplot as plt
import numpy as np
import procrustes


A = np.random.random((3, 3))
B = np.dot(A, np.array([[np.cos(2), -np.sin(2), 0], [np.sin(2), np.cos(2), 0], [0, 0, 1]])) * 10 + 4

r = procrustes.orthogonal(A, B, scale=True, translate=True)

_, a_c = procrustes.utils._translate_array(A, None)
_, a_s = procrustes.utils._scale_array(A + a_c, None)

_, b_c = procrustes.utils._translate_array(B, None)
_, b_s = procrustes.utils._scale_array(B + b_c, None)

transformation_matrix = r.t

AT = np.dot((A + a_c) * a_s, r.t)
nb = (B + b_c) * b_s
ATB = AT / b_s - b_c
print(f"{AT=}")
print(f"{r.new_b=}")
print(f"{AT / b_s - b_c=}")
print(f"{B=}")

fig = plt.figure()
ax = fig.add_subplot(projection="3d")
ax.plot(AT[:, 0], AT[:, 1], AT[:, 2], 'x', mew=3)
ax.plot(nb[:, 0], nb[:, 1], nb[:, 2], 'x', mew=3)

fig = plt.figure()
ax = fig.add_subplot(projection="3d")
ax.plot(ATB[:, 0], ATB[:, 1], ATB[:, 2], 'x', mew=3)
ax.plot(B[:, 0], B[:, 1], B[:, 2], 'x', mew=3)

plt.show()
```

**Reply by FanwangM** [reply]
It's great to know that you have solved the problem. 
Please feel free to tag me if you need more help or have new questions.
@studywolf
